src/api/cli/integration/integration_test.go

```diff
@@ -1648,3 +1822,111 @@ func getWalletOutputs(t *testing.T, walletPath string) visor.ReadableOutputs {
 
 	return wltOutput.Outputs.HeadOutputs
 }
+
+func TestStableWalletHistory(t *testing.T) {
+	if !doStable(t) {
+		return
+	}
+
+	_, clean := createTempWalletFile(t)
+	defer clean()
+
+	output, err := exec.Command(binaryPath, "walletHistory").CombinedOutput()
+	require.NoError(t, err)
+
+	var history []cli.AddrHistory
+	err = json.NewDecoder(bytes.NewReader(output)).Decode(&history)
+	require.NoError(t, err)
+
+	var expect []cli.AddrHistory
+	loadGoldenFile(t, "wallet-history.golden", TestData{history, &expect})
+	require.Equal(t, expect, history)
+}
+
+func TestLiveWalletHistory(t *testing.T) {
+	if !doLive(t) {
+		return
+	}
+
+	output, err := exec.Command(binaryPath, "walletHistory").CombinedOutput()
+	require.NoError(t, err)
+	var his []cli.AddrHistory
+	err = json.NewDecoder(bytes.NewReader(output)).Decode(&his)
+	require.NoError(t, err)
+	// Confirms that the wallet has history.
+	require.True(t, len(his) > 0)
+}
+
+func TestCheckDB(t *testing.T) {
```

Comment:
Is it possible for `TestLiveCheckDB` to check the user's db?

Reply:
Yes, I will add it.

Reply:
Hmm, seems can't. The live test loads database with writing access, we can't open it in another progress even try with readonly option.

Reply:
oh, it would have to be a separate test with no node running, its fine

Reply:
We should make a pre-release testing checklist, with instructions to run the live tests, and an additional step to run checkdb on a synced database